Incorrect C408 rule warning for dict with keyword arguments

The [C408 rule](https://github.com/adamchainz/flake8-comprehensions#c408-unnecessary-dictlisttuple-call---rewrite-as-a-literal) seems to fail on a dict with keyword arguments, but from the description of the rule, it should not fail.

```
$ cat scratch.py 
a = dict(one=1, two=2, three=3)

$ flake8 --version
3.8.2 (flake8-bugbear: 20.1.4, flake8-comprehensions: 3.2.2, mccabe: 0.6.1, pycodestyle: 2.6.0, pyflakes: 2.2.0) CPython 3.8.2
on Darwin

$ flake8 ~/Library/Preferences/PyCharmCE2019.3/scratches/scratch.py 
scratch.py:1:5: C408 Unnecessary dict call - rewrite as a literal.
```

--- a/HISTORY.rst
+++ b/HISTORY.rst
@@ -2,6 +2,9 @@
 History
 =======
 
+* Made ``C408`` only apply when no arguments are passed to
+  ``dict``/``list``/``tuple``.
+
 3.2.2 (2020-01-20)
 ------------------
 

--- a/README.rst
+++ b/README.rst
@@ -130,9 +130,10 @@ The list of builtins that are checked for are:
 C408: Unnecessary (dict/list/tuple) call - rewrite as a literal.
 ~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~~
 
-It's slower to call e.g. ``dict()`` than using the empty literal, because the
-name ``dict`` must be looked up in the global scope in case it has been
-rebound. Same for the other two basic types here. For example:
+It's slower to call these types than to construct them with literals, e.g.
+``dict()`` is slower than ``{}``, because the name ``dict`` must be looked up
+in the global scope in case it has been rebound, and it includes the overhead
+of a function call. Same for the other two builtin types here. For example:
 
 * Rewrite ``dict()`` as ``{}``
 * Rewrite ``list()`` as ``[]``

--- a/src/flake8_comprehensions.py
+++ b/src/flake8_comprehensions.py
@@ -187,9 +187,8 @@ class ComprehensionChecker:
 
                 elif (
                     num_positional_args == 0
-                    and not has_star_args(node)
-                    and not has_keyword_args(node)
                     and node.func.id in ("tuple", "list", "dict")
+                    and len(node.keywords) == 0
                 ):
                     yield (
                         node.lineno,
@@ -319,14 +318,6 @@ class ComprehensionChecker:
                         self.messages["C416"].format(type=comp_type[node.__class__]),
                         type(self),
                     )
-
-
-def has_star_args(call_node):
-    return any(isinstance(a, ast.Starred) for a in call_node.args)
-
-
-def has_keyword_args(call_node):
-    return any(k.arg is None for k in call_node.keywords)
 
 
 comp_type = {

--- a/tests/test_flake8_comprehensions.py
+++ b/tests/test_flake8_comprehensions.py
@@ -618,6 +618,12 @@ def test_C408_pass_6(flake8dir):
     assert result.out_lines == []
 
 
+def test_C408_pass_7(flake8dir):
+    flake8dir.make_example_py("dict(a=1)")
+    result = flake8dir.run_flake8()
+    assert result.out_lines == []
+
+
 def test_C408_fail_1(flake8dir):
     flake8dir.make_example_py("tuple()")
     result = flake8dir.run_flake8()
@@ -636,14 +642,6 @@ def test_C408_fail_2(flake8dir):
 
 def test_C408_fail_3(flake8dir):
     flake8dir.make_example_py("dict()")
-    result = flake8dir.run_flake8()
-    assert result.out_lines == [
-        "./example.py:1:1: C408 Unnecessary dict call - rewrite as a literal."
-    ]
-
-
-def test_C408_fail_4(flake8dir):
-    flake8dir.make_example_py("dict(a=1)")
     result = flake8dir.run_flake8()
     assert result.out_lines == [
         "./example.py:1:1: C408 Unnecessary dict call - rewrite as a literal."

